lib/jsdom/living/nodes/XMLDocument-impl.js

```diff
@@ -1,4 +1,9 @@
 "use strict";
 const DocumentImpl = require("./Document-impl").implementation;
 
-exports.implementation = class XMLDocumentImpl extends DocumentImpl {};
+exports.implementation = class XMLDocumentImpl extends DocumentImpl {
+  constructor(globalObject, args, privateData = {}) {
+    // XMLDocument is always in "xml" parsing mode:
+    super(globalObject, args, { ...privateData, options: { ...privateData.options, parsingMode: "xml" } });
```

Comment:
I'm still a bit unclear why this is necessary. There is no public constructor in the IDL for `XMLDocument`, so the only way this could ever have the wrong parsing mode is if some other part of the code passed it. But, then we should fix the other part of the code, right?

So I think we can just undo all changes to this file. I will push a commit doing so momentarily.